plugins/utils/hardtime.nix

```diff
@@ -0,0 +1,169 @@
+          Allow different keys to reset the count.
+        '';
+
+        enabled = helpers.defaultNullOpts.mkBool true ''
+          Whether the plugin in enabled by default or not.
+        '';
+
+        resettingKeys = helpers.mkNullOrOption (with types; attrsOf (listOf str)) ''
+          Keys in what modes that reset the count.
+
+          default:
+          ```nix
+          {
+            "1" = [ "n" "x" ];
+            "2" = [ "n" "x" ];
+            "3" = [ "n" "x" ];
+            "4" = [ "n" "x" ];
+            "5" = [ "n" "x" ];
+            "6" = [ "n" "x" ];
+            "7" = [ "n" "x" ];
+            "8" = [ "n" "x" ];
+            "9" = [ "n" "x" ];
+            "c" = [ "n" ];
+            "C" = [ "n" ];
+            "d" = [ "n" ];
+            "x" = [ "n" ];
+            "X" = [ "n" ];
+            "y" = [ "n" ];
+            "Y" = [ "n" ];
+            "p" = [ "n" ];
+            "P" = [ "n" ];
+          }
+          ```
+        '';
+
+        restrictedKeys = helpers.mkNullOrOption (with types; attrsOf (listOf str)) ''
+          Keys in what modes triggering the count mechanism.
+
+          default:
+          ```nix
+          {
+            "h" = [ "n" "x" ];
+            "j" = [ "n" "x" ];
+            "k" = [ "n" "x" ];
+            "l" = [ "n" "x" ];
+            "-" = [ "n" "x" ];
+            "+" = [ "n" "x" ];
+            "gj" = [ "n" "x" ];
+            "gk" = [ "n" "x" ];
+            "<CR>" = [ "n" "x" ];
+            "<C-M>" = [ "n" "x" ];
+            "<C-N>" = [ "n" "x" ];
+            "<C-P>" = [ "n" "x" ];
+          }
+          ```
+        '';
+
+        restrictionMode = helpers.defaultNullOpts.mkEnumFirstDefault ["block" "hint"] ''
+          The behavior when `restricted_keys` trigger count mechanism.
+        '';
+
+        disabledKeys = helpers.mkNullOrOption (with types; attrsOf (listOf str)) ''
+          Keys in what modes are disabled.
+
+          default:
+          ```nix
+          {
+            "<Up>" = [ "" "i" ];
+            "<Down>" = [ "" "i" ];
+            "<Left>" = [ "" "i" ];
+            "<Right>" = [ "" "i" ];
+          }
+          ```
+        '';
+
+        disabledFiletypes = helpers.mkNullOrOption (with types; listOf str) ''
+          `hardtime.nvim` is disabled under these filetypes.
+
+          default:
+          ```nix
+          ["qf" "netrw" "NvimTree" "lazy" "mason"]
+          ```
+        '';
+
+        hints = helpers.mkNullOrOption (with types;
+          attrsOf (submodule {
+            options = {
+              message = lib.mkOption {
+                description = "Hint message to be displayed.";
+                type = with types; either str helpers.rawType;
+                default = {};
+              };
+
+              length = lib.mkOption {
+                description = "The length of actual key strokes that matches this pattern.";
+                type = types.int;
+                default = {};
+              };
```

Comment:
Are those options mandatory ? If so, `default` should not be set.
Else, why don't you use `mkNullOrOption` or `defaultNullOpts.mk__` ?